fileAttributeCaching/src/main/kotlin/com/pkware/java/nio/file/fileAttributeCaching/FileAttributeCachingPath.kt

```diff
@@ -0,0 +1,201 @@
+package com.pkware.java.nio.file.fileAttributeCaching
+
+import com.pkware.java.nio.file.forwarding.ForwardingPath
+import java.nio.file.FileSystem
+import java.nio.file.Path
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.attribute.DosFileAttributes
+import java.nio.file.attribute.PosixFileAttributes
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+
+private const val CACHE_PRESERVATION_SECONDS: Long = 5
```

Comment:
Why 5 seconds?

Reply:
added a short explination blurb but tldr marius thought this value was good and most file operations complete in this time for our unpacker use cases